api/analyzers/csharp/analyzer.py

```
@@ -0,0 +1,137 @@
+
+    def _captures(self, pattern: str, node: Node) -> dict:
+        """Run a tree-sitter query and return captures dict."""
+        query = self.language.query(pattern)
+        cursor = QueryCursor(query)
+        return cursor.captures(node)
+
+    def add_dependencies(self, path: Path, files: list[Path]):
+        if Path(f"{path}/temp_deps_cs").is_dir():
+            return
+        if Path(f"{path}").glob("*.csproj") or Path(f"{path}").glob("*.sln"):
+            subprocess.run(["dotnet", "restore"], cwd=str(path))
```

Ruff (0.15.1)

[error] Starting a process with a partial executable path

(S607)